### ISSUE TITLE
Gaussian Mixture Model: Tensor had Inf values

### ISSUE THREAD

**Issue by demodw** [blei-lab/edward#477]
Hi,

I've used the example code for a GMM implemented in Tensorflow (https://github.com/blei-lab/edward/blob/master/examples/tf_mixture_gaussian.py), but when I increase the number of dimensions to 128 I get the following error,

InvalidArgumentError (see above for traceback):  : Tensor had Inf values
	 [[Node: VerifyFinite_7/CheckNumerics = CheckNumerics[T=DT_FLOAT, _class=["loc:@stack_7"], message="", _device="/job:localhost/replica:0/task:0/gpu:0"](stack_7)]]
	 [[Node: Neg/_669 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_16417_Neg", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

I modified the toy data set to just give some values from a normal(0,1) and then multiplied some of it by 2,

N = 500
K = 2
D = 128
x_train = np.random.normal(0, 1, size=(N, D))
x_train[100:,] *= 2

Any idea what's going on?

**Comment by mahatosourav91** [comment]
Hi, I am facing the same issue. I have modified the example given [here](https://github.com/blei-lab/edward/blob/master/examples/tf_mixture_gaussian.py), but I replaced the toy dataset with my dataset. 
My data is a numpy array of shape (500,198)
Also in my case `N=500, K=200, D=198`. 
Please look into it

**Comment by dustinvtran** [comment]
You may have to play with `n_samples`, the initialization of the variational families, and the `optimizer`. Dirichlet's can be quite tough with the generic score function gradient method which is used in `KLqp`.

**Comment by demodw** [comment]
I get the same error with 1000 samples (inference.run(n_iter=2500, n_samples=1000, n_minibatch=20)), using gradientdescent, adadelta, adagrad, but no luck.

The same problem is running fine using the sklearn implementation (http://scikit-learn.org/stable/modules/generated/sklearn.mixture.BayesianGaussianMixture.html), so I guess this may just be an illposed problem.
